src/compiler/checker.ts

```diff
@@ -44752,44 +44752,63 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
-                                return evaluate(declaration.initializer, declaration);
-                            }
-                        }
+                return evaluateEntityNameExpression(expr as EntityNameExpression, location);
+            case SyntaxKind.ElementAccessExpression:
+                return evaluateElementAccessExpression(expr as ElementAccessExpression, location);
+        }
+        return undefined;
+    }
+
+    function evaluateEntityNameExpression(expr: EntityNameExpression, location?: Declaration) {
+        if (isIdentifier(expr) && isInfinityOrNaNString(expr.escapedText) && 
+            (resolveEntityName(expr, SymbolFlags.Value, /*ignoreErrors*/ true) === getGlobalSymbol(expr.escapedText, SymbolFlags.Value, /*diagnostic*/ undefined))) {
+            return +(expr.escapedText);
+        }
+        if (isEntityNameExpression(expr)) {
+            const symbol = resolveEntityName(expr, SymbolFlags.Value, /*ignoreErrors*/ true);
+            if (!symbol) return undefined;
+            if (symbol.flags & SymbolFlags.EnumMember) {
+                if (!symbol.valueDeclaration) return undefined;
+                const enumMember = symbol.valueDeclaration as EnumMember;
+                if (location) {
+                    if (compilerOptions.isolatedDeclarations && location.parent !== enumMember.parent) {
+                        return undefined;
                     }
+                    return evaluateEnumMember(expr, symbol, location);
                 }
-                break;
-            case SyntaxKind.ElementAccessExpression:
-                const root = (expr as ElementAccessExpression).expression;
-                if (isEntityNameExpression(root) && isStringLiteralLike((expr as ElementAccessExpression).argumentExpression)) {
-                    const rootSymbol = resolveEntityName(root, SymbolFlags.Value, /*ignoreErrors*/ true);
-                    if (rootSymbol && rootSymbol.flags & SymbolFlags.Enum) {
-                        const name = escapeLeadingUnderscores(((expr as ElementAccessExpression).argumentExpression as StringLiteralLike).text);
-                        const member = rootSymbol.exports!.get(name);
-                        if (member) {
-                            return location ? evaluateEnumMember(expr, member, location) : getEnumMemberValue(member.valueDeclaration as EnumMember);
+                return getEnumMemberValue(enumMember);
+            }
+            if (compilerOptions.isolatedDeclarations && isConstantVariable(symbol)) {
+                const declaration = symbol.valueDeclaration;
+                if (declaration && isVariableDeclaration(declaration) && !declaration.type && declaration.initializer && (!location || declaration !== location && isBlockScopedNameDeclaredBeforeUse(declaration, location))) {
+                    return evaluate(declaration.initializer, declaration);
+                }
+            }
+        }

```

Comment:
I am a bit reticent to not compute the value at all if we are in isolated declarations. I mean we can compute the value, it's just we can't emit it..

Reply:
I looked again and think for a while, but I feel the changes required for preserving the value, isn't worth it. Would you strongly push for this?

Reply:
No. It's fine